test/workflow-orchestration.ts

```diff
@@ -138,6 +138,70 @@ describe('Workflow chaining for a collection configured for swot reprojection an
     });
   });
 
+  describe('when making a request and the job fails while in progress', function () {
+    const reprojectAndZarrQuery = {
+      maxResults: 3,
+      outputCrs: 'EPSG:4326',
+      interpolation: 'near',
+      scaleExtent: '0,2500000.3,1500000,3300000',
+      scaleSize: '1.1,2',
+      format: 'application/x-zarr',
+      turbo: true,
+    };
+
+    hookRangesetRequest('1.0.0', collection, 'all', { query: reprojectAndZarrQuery });
+    hookRedirect('joe');
+
+    describe('when checking for a query-cmr work item', function () {
```

Comment:
It's a little confusing that you are testing to see if query-cmr gets executed here instead of just executing it to get the side effect that you want (creation of the work items for the next step). In fact, query-cmr is not really special and should have the same behavior as any other service when the job is canceled (shouldn't get any work items), so you could just use it instead of a subsequent service. Not a hug deal, just a bit confusing to see a 'success' test case inside a 'job fails' `describe`.